sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,264 @@
+          )
+        : parentProcessSummary.warn(
+            `Invoice ${pendingInvoicePayload.invoiceNumber} send to CAS did not succeed. Reason: ${response.casReturnedMessages}.`,
+          );
+      await this.dataSource.getRepository(CASInvoice).update(
+        {
+          id: pendingInvoice.id,
+        },
+        {
+          invoiceStatus: CASInvoiceStatus.Sent,
+          invoiceStatusUpdatedOn: new Date(),
+          modifier: { id: this.systemUsersService.systemUser.id },
+        },
+      );
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        if (error.name === CAS_BAD_REQUEST) {
+          parentProcessSummary.warn("Known CAS error while sending invoice.");
+          await this.processBadRequestErrors(
+            pendingInvoice.id,
+            parentProcessSummary,
+            error.objectInfo as string[],
+            this.systemUsersService.systemUser.id,
+          );
+        }
+      } else {
+        parentProcessSummary.error(
+          "Unexpected error while sending invoice to CAS.",
+          error,
+        );
+        return null;
+      }
+    }
+  }
+
+  /**
+   * Process bad request errors from CAS API during invoice sending.
+   * @param invoiceId invoice id.
+   * @param summary summary for logging.
+   * @param error error object.
+   * @param auditUserId audit user id.
+   */
+  async processBadRequestErrors(
+    invoiceId: number,
+    summary: ProcessSummary,
+    error: string[],
+    auditUserId: number,
+  ): Promise<void> {
+    summary.warn("A known error occurred during processing.");
+    const now = new Date();
+    const auditUser = { id: auditUserId } as User;
+    try {
+      await this.casInvoiceRepo.update(
+        {
+          id: invoiceId,
+        },
+        {
+          invoiceStatus: CASInvoiceStatus.ManualIntervention,
+          invoiceStatusUpdatedOn: now,
+          modifier: auditUser,
+          errors: error,
+        },
+      );
+      summary.info(
+        `Set invoice status to ${CASInvoiceStatus.ManualIntervention} due to Error(s):${error}.`,
+      );
+    } catch (error: unknown) {
+      summary.error(
+        `Failed to update invoice status to ${CASInvoiceStatus.ManualIntervention}.`,
+        error,
+      );
+    }
+  }
+
+  /**
+   * Generate the pending invoice payload.
+   * @param pendingInvoice pending invoice.
+   * @returns pending invoice payload.
+   */
+  private getPendingInvoicePayload(
+    pendingInvoice: CASInvoice,
+  ): PendingInvoicePayload {
+    // Fixed values as constants.
+    const INVOICE_TYPE = "Standard";
+    const INVOICE_AMOUNT = 0;
+    const PAY_GROUP = "GEN GLP";
+    const REMITTANCE_CODE = "01";
+    const SPECIAL_HANDLING = "N";
+    const TERMS = "Immediate";
+    const EMPTY_STRING = "";
+    const CURRENCY_CODE = "CAD";
+    const INVOICE_LINE_TYPE = "Item";
+
+    // Get invoice line details.
+    const invoiceLineDetails: InvoiceLineDetail[] =
```

Comment:
Can we add an order by to the query to ensure the invoice details will always follow the same order? For instance, order by award code, then by operation?